flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/web/SetupInfoServlet.java

```diff
@@ -127,25 +134,42 @@ private void writeTaskmanagers(HttpServletResponse resp) throws IOException {
 				objInner.put("physicalMemory", instance.getResources().getSizeOfPhysicalMemory() >>> 20);
 				objInner.put("freeMemory", instance.getResources().getSizeOfJvmHeap() >>> 20);
 				objInner.put("managedMemory", instance.getResources().getSizeOfManagedMemory() >>> 20);
+				objInner.put("instanceID", instance.getId());
 				array.put(objInner);
 			}
 			catch (JSONException e) {
 				LOG.warn("Json object creation failed", e);
 			}
-			
+
 		}
 		try {
 			obj.put("taskmanagers", array);
 		} catch (JSONException e) {
 			LOG.warn("Json object creation failed", e);
 		}
-		
+
 		PrintWriter w = resp.getWriter();
 		w.write(obj.toString());
 	}
-	
+
+	private void writeStackTraceOfTaskManager(String instanceIdStr, HttpServletResponse resp) throws IOException {
```

Comment:
The `RequestStackTrace` message may fail, if the task manager is not reachable.

I suggest to surround this block with try / catch(Throwable) and forward the error message to the web client.

The response JSON may then have two fields: "errorMessage" and "stackTrace". If "errorMessage" is defined, display the message, otherwise print the stack trace.